### PR TITLE
etcd: Updated to v3.5.5

### PR DESCRIPTION
**Pull request by dims**
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?

/kind bug
/kind cleanup

#### What this PR does / why we need it:
Update etcd to [v3.5.5](https://github.com/etcd-io/etcd/releases/tag/v3.5.5).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
etcd: Update to v3.5.5
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
We can `/retest ` after https://github.com/kubernetes/k8s.io/pull/4222 is merged.

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -476,10 +476,10 @@ var (
...
 		20: "3.4.13-0",
 		21: "3.4.13-0",
-		22: "3.5.4-0",
-		23: "3.5.4-0",
-		24: "3.5.4-0",
-		25: "3.5.4-0",
+		22: "3.5.5-0",
+		23: "3.5.5-0",
+		24: "3.5.5-0",
+		25: "3.5.5-0",
```
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go:482]: /approve
for kubeadm.
thanks

**Comment by ahrtr** [comment]
Please note that there are also bug fixes on [clientv3](https://github.com/etcd-io/etcd/tree/main/client/v3) in 3.5.5, so you might want to bump etcd-client dependencies as well. e.g go.mod/go.sum

**Comment by warmchang** [comment]
ref: https://github.com/kubernetes/kubernetes/pull/112545

**Pull request merged**